seastar/src/timer.rs

```diff
@@ -12,11 +12,36 @@ mod ffi {
         fn steady_clock_now() -> i64;
 
         fn new_sct() -> UniquePtr<steady_clock_timer>;
+
+        unsafe fn sct_set_callback(
+            timer: Pin<&mut steady_clock_timer>,
+            callback: *mut u8, // u8 is a substitute for c_void that isn't supported by cxx.
+            caller: unsafe fn(*mut u8),
+            dropper: unsafe fn(*mut u8),
+        );
     }
 }
 
 use ffi::*;
 
+fn caller<Func: FnMut()>(raw_func: *mut u8) {
+    unsafe { (raw_func as *mut Func).as_mut().unwrap()() }
+}
+
+const fn get_caller<Func: FnMut()>(_: &Func) -> fn(*mut u8) {
+    caller::<Func>
+}
+
+fn dropper<Func: FnMut()>(raw_func: *mut u8) {
+    unsafe {
+        let _ = Box::from_raw(raw_func as *mut Func);
+    }
+}
+
+const fn get_dropper<Func: FnMut()>(_: &Func) -> fn(*mut u8) {
+    dropper::<Func>
+}
```

Comment:
Nit: the `FnMut()` bounds here are unnecessary.

Reply:
Good point, I fixed it.